### PR TITLE
Remove Models-TypeSpec mockapis

### PR DESCRIPTION
**Pull request by mcgallan**
Add test for type/property/optional and type/model/visibility, remove Models-TypeSpec in CadlRanchMockApi and related tests

### CODE REVIEW

File: test/CadlRanchProjectsNonAzure.Tests/type-property-optionality.cs [hunk 1]
```diff
@@ -14,6 +14,8 @@ namespace CadlRanchProjectsNonAzure.Tests
 {
     public class TypePropertyOptionalTests : CadlRanchNonAzureTestBase
     {
+        private static readonly DateTimeOffset PlainDateData = new DateTimeOffset(2022, 12, 12, 0, 0, 0, 0, new TimeSpan());
+        private static readonly TimeSpan PlainTimeData = new TimeSpan(13, 06, 12);
```
Comment (pshao25) [inline test/CadlRanchProjectsNonAzure.Tests/type-property-optionality.cs:18]: Do we need this?

File: test/CadlRanchProjects.Tests/type-model-visibility.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace CadlRanchProjects.Tests
+{
+    public class TypeModelVisibilityTests : CadlRanchTestBase
+    {
+        [Test]
+        public Task Models_ReadOnlyRoundTrip() => Test(async (host) =>
+        {
+            var response = await new VisibilityClient(host, null).PutReadOnlyModelAsync(new ReadOnlyModel()
+            {
```
Comment (pshao25) [inline test/CadlRanchProjects.Tests/type-model-visibility.cs:24]: Do we need curly braces?

File: test/CadlRanchProjects.Tests/type-property-optionality.cs [hunk 3]
```diff
@@ -14,6 +15,8 @@ namespace CadlRanchProjects.Tests
 {
     public class TypePropertyOptionalTests : CadlRanchTestBase
     {
+        private static readonly DateTimeOffset PlainDateData = new DateTimeOffset(2022, 12, 12, 0, 0, 0, 0, new TimeSpan());
```
Comment (pshao25) [inline test/CadlRanchProjects.Tests/type-property-optionality.cs:18]: Same

File: test/CadlRanchMockApis/tsconfig.json [hunk 4]
```diff
@@ -29,4 +29,3 @@
     },
     "include": ["src/**/*.ts", "test/**/*.ts"]
 }
-
```
Comment (pshao25) [inline test/CadlRanchMockApis/tsconfig.json:32]: Revert unnecessary change

File: test/AutoRest.TestServer.Tests/LowLevel/DecoratorTests.cs [hunk 5]
```diff
@@ -25,26 +24,5 @@ public class DecoratorTests
             Assert.IsNotNull(attribute);
             Assert.AreEqual("deprecated for test", ((ObsoleteAttribute)attribute).Message);
         }
-
```
Comment (pshao25) [inline test/AutoRest.TestServer.Tests/LowLevel/DecoratorTests.cs:28]: Why we remove this test?

Comment (mcgallan) [reply test/AutoRest.TestServer.Tests/LowLevel/DecoratorTests.cs:28]: Because the models mockapi-related operations have been removed, there are no longer any operations handled with @deprecated(""). This test case has lost its object to test.

File: test/CadlRanchProjectsNonAzure.Tests/type-property-optionality.cs (hunk 1 above)
Comment (mcgallan) [reply test/CadlRanchProjectsNonAzure.Tests/type-property-optionality.cs:18]: removed

File: test/CadlRanchProjects.Tests/type-property-optionality.cs (hunk 3 above)
Comment (mcgallan) [reply test/CadlRanchProjects.Tests/type-property-optionality.cs:18]: removed

File: test/CadlRanchProjects.Tests/type-model-visibility.cs (hunk 2 above)
Comment (mcgallan) [reply test/CadlRanchProjects.Tests/type-model-visibility.cs:24]: removed

File: test/CadlRanchMockApis/tsconfig.json (hunk 4 above)
Comment (mcgallan) [reply test/CadlRanchMockApis/tsconfig.json]: fixed

File: test/AutoRest.TestServer.Tests/LowLevel/DecoratorTests.cs (hunk 5 above)
Comment (pshao25) [reply test/AutoRest.TestServer.Tests/LowLevel/DecoratorTests.cs:28]: Your justification is not correct. The object it is leveraging becomes unavailable is not the reason you could remove. You should understand what this test is for, before we decide whether it could be deleted, or move to another target.

File: test/CadlRanchProjects.Tests/type-property-optionality.cs (hunk 3 above)
Comment (pshao25) [reply test/CadlRanchProjects.Tests/type-property-optionality.cs:18]: What did you remove?

Comment (mcgallan) [reply test/CadlRanchProjects.Tests/type-property-optionality.cs:18]: This part of the code is used in Cadl Ranch and should be removed in nonazure. I mixed up the replies for these two parts. The related changes have been correctly applied.

**Review by pshao25 — APPROVED** [review]
